### PR TITLE
The Graph Docs x Nextra

### PR DESCRIPTION
**Pull request by dimaMachina**
<s>I'll rebase to main once `saihaj/move-to-monorepo` will be merged</s>

Follow up items
- https://github.com/graphprotocol/docs/issues/310
- https://github.com/graphprotocol/docs/issues/311

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -21,19 +21,19 @@
...
     "husky": "^8.0.3",
     "lint-staged": "^13.1.2",
     "prettier": "^2.8.4",
     "typescript": "4.9.5"
   },
-  "pnpm": {
-    "overrides": {
-      "@types/react": "^17.0.52"
-    }
-  },
```
Comment (B2o5T) [inline package.json:34]: we use react 18

File: website/navigation/utils.ts [hunk 2]
```diff
@@ -37,8 +41,14 @@ export const getNavItems = async (locale: AppLocale = defaultLocale): Promise<Na
...
-                const { frontmatter }: { frontmatter?: Frontmatter | ((locale: AppLocale) => Frontmatter) } =
-                  await import(`../pages/${fileToTry}`)
+                let frontmatter: Frontmatter | ((locale: AppLocale) => Frontmatter)
+                if (fileToTry.endsWith('.mdx')) {
+                  const { data } = grayMatter(await readFile(join(process.cwd(), 'pages', fileToTry), 'utf8'))
+                  frontmatter = data
+                } else {
+                  const mod = await import(`../pages/${fileToTry}`)
+                  frontmatter = mod.frontmatter
+                }
```
Comment (B2o5T) [inline website/navigation/utils.ts:51]: the temporal solution, after merging this PR I'll make refactoring to use Nextra's `_meta.json` so this file will be completely removed

File: website/next.config.mjs [hunk 3]
```diff
@@ -13,43 +8,52 @@ const env = {
...
+import { getNavItems } from '@/navigation'
+
+${content}
+
+export const getStaticProps = async () => {
+  const navItems = await getNavItems('${route.split('/')[1]}')
+  return {
+    props: { navItems }
+  }
+}`
```
Comment (B2o5T) [inline website/next.config.mjs:29]: the temporal solution, once we'll use `_meta.json` I'll remove it and we'll have the possibility that every page could have their`getStaticProps`

File: website/next.config.mjs [hunk 4]
```diff
@@ -13,43 +8,52 @@ const env = {
...
-export default withMDX({
+export default withNextra({
   experimental: {
     // Fix scroll restoration (see https://github.com/vercel/next.js/issues/37893#issuecomment-1221335543)
     scrollRestoration: true,
   },
-
   env,
-  pageExtensions: ['tsx', 'mdx'],
+  pageExtensions: ['tsx'],
```
Comment (B2o5T) [inline website/next.config.mjs:41]: don't need it because nextra already concat `md` and `mdx` extensions

File: website/package.json [hunk 5]
```diff
@@ -14,46 +14,30 @@
...
-    "acorn": "^8.8.2",
-    "acorn-jsx": "^5.3.2",
-    "eslint": "^8.35.0",
     "fast-xml-parser": "^4.1.3",
-    "remark-frontmatter": "^4.0.1",
-    "remark-mdx-frontmatter": "^2.1.1",
-    "serialize-as-code": "^2.0.2",
-    "unist-util-visit": "^4.1.2",
-    "url": "^0.11.0"
+    "gray-matter": "^4.0.3"
```
Comment (B2o5T) [inline website/package.json:40]: temporal dep for parsing frontmatter

**Review by saihaj — APPROVED** [review]

**Review by benface — COMMENTED** [review]
I'm a bit confused. I'm trying to run this locally, but am getting an error after running `pnpm install` and `pnpm dev`:
> next: command not found

**Comment by B2o5T** [comment]
> I'm a bit confused. I'm trying to run this locally, but am getting an error after running pnpm install and pnpm dev:

@benface now the repository is a monorepo, run `pnpm --filter @graphprotocol/docs dev` instead

**Comment by benface** [comment]
Thank you @B2o5T! This is really impressive work. I was expecting the diff to be a lot larger than this.

I found a couple issues, so I made my own PR with your branch as a base to address them: #319. Please see the commit messages for explanations. I'm sure you'll find better ways to fix the issues I found, so feel free to work in my branch, merge it directly, or give me feedback.

**Comment by B2o5T** [comment]
> This is really impressive work.
Thanks!

> I was expecting the diff to be a lot larger than this.
I didn't want to put too many things in simple PR, so I will continue on migration in the following PRs

- use `_meta.json`s files for building sidebar links
- export `graph` theme as external package
- improve code-blocks with copy to clipboard button

**Comment by benface** [comment]
> improve code-blocks with copy to clipboard button

Actually, for that one, I was planning on using our `<Code.Block>` component from `@edgeandnode/components`. It will make code blocks consistent with how they appear elsewhere, e.g. on [the blog](https://thegraph.com/blog/improve-subgraph-performance-reduce-eth-calls/#example-schema) (note that there is no "copy" button here, but there will be once we update the component).

**Comment by B2o5T** [comment]
> Actually, for that one, I was planning on using our <Code.Block>

ok I'll leave it for you

**Pull request merged**